src/lib.rs

```diff
@@ -667,16 +665,54 @@ impl Config {
                     }
                     cmd.arg("-AWin32");
                 } else {
-                    panic!("unsupported msvc target: {}", target);
+                    panic!("unsupported msvc target: {target}");
                 }
             }
-        } else if target.contains("darwin") && !self.defined("CMAKE_OSX_ARCHITECTURES") {
-            if target.contains("x86_64") {
-                cmd.arg("-DCMAKE_OSX_ARCHITECTURES=x86_64");
-            } else if target.contains("aarch64") {
-                cmd.arg("-DCMAKE_OSX_ARCHITECTURES=arm64");
-            } else {
-                panic!("unsupported darwin target: {}", target);
+        } else if target.contains("apple") {
+            if !self.defined("CMAKE_OSX_ARCHITECTURES") {
+                if target.contains("x86_64") {
+                    cmd.arg("-DCMAKE_OSX_ARCHITECTURES=x86_64");
+                } else if target.contains("aarch64") {
+                    cmd.arg("-DCMAKE_OSX_ARCHITECTURES=arm64");
+                } else if target.contains("i686") {
+                    cmd.arg("-DCMAKE_OSX_ARCHITECTURES=i686");
+                } else if target.contains("arm64e") {
+                    cmd.arg("-DCMAKE_OSX_ARCHITECTURES=arm64e");
+                } else if target.contains("i386") {
+                    cmd.arg("-DCMAKE_OSX_ARCHITECTURES=i386");
+                } else if target.contains("armv7s") {
+                    cmd.arg("-DCMAKE_OSX_ARCHITECTURES=armv7s");
+                } else if target.contains("armv7k") {
+                    cmd.arg("-DCMAKE_OSX_ARCHITECTURES=armv7k");
+                } else if target.contains("arm64_32") {
+                    cmd.arg("-DCMAKE_OSX_ARCHITECTURES=arm64_32");
+                } else {
+                    panic!("unsupported apple target: {target}");
```

Comment:
Nit: Maybe we can make this simpler, and remove the panic? See bootstrap's implementation:
https://github.com/rust-lang/rust/blob/718ddf660e6a1802c39b4962cf7eaa4db57025ef/src/bootstrap/src/core/build_steps/llvm.rs#L698-L706